RELEASE.md

```diff
@@ -61,53 +61,19 @@ If this is a .0 release:
 RELEASE-TEMPLATE-END
```

Comment:
this whole release-template thing has moved into a new file. please remove it from here. See https://github.com/eclipse/che/blob/master/.github/ISSUE_TEMPLATE/release.md for the up to date issue template.